From: Carlos Garnacho <carlosg@gnome.org>
Date: Mon, 30 Nov 2020 19:00:31 +0100
Subject: backends/x11: Use XkbBuildCoreState the right way around

Pass parameters in the correct order, and don't let it clamp button
modifiers away (Since this macro does "state & 0xff").

Fixes: https://gitlab.gnome.org/GNOME/mutter/-/issues/1559
Part-of: <https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/1617>
---
 src/backends/x11/meta-seat-x11.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/backends/x11/meta-seat-x11.c b/src/backends/x11/meta-seat-x11.c
index 2381ddf..4fa8675 100644
--- a/src/backends/x11/meta-seat-x11.c
+++ b/src/backends/x11/meta-seat-x11.c
@@ -1521,7 +1521,7 @@ translate_state (XIButtonState   *button_state,
     }
 
   if (group_state)
-    state = XkbBuildCoreState (group_state->effective, state);
+    state |= XkbBuildCoreState (0, group_state->effective);
 
   return state;
 }
